modules/document_repository/php/files.class.inc

```diff
@@ -0,0 +1,497 @@
+            return (new \LORIS\Http\Response())
+                ->withHeader("Content-Type", "text/plain")
+                ->withStatus(200)
+                ->withBody(
+                    new \LORIS\Http\StringStream(
+                        json_encode("delete a file")
+                    )
+                );
+          break;
+        case "GET":
+            $record = basename($request->getUri()->getPath());
+            if (!is_numeric($record)) {
+                $file = __DIR__ . "/../user_uploads/$record";
+                return (new \LORIS\Http\Response())
+                ->withHeader(
+                    'Content-Disposition',
+                    'attachment; filename=' . basename($file)
+                )
+                ->withHeader('Content-Transfer-Encoding', 'Binary')
+                ->withBody(new \LORIS\Http\StringStream(readfile($file)));
+
+            }
+            return (new \LORIS\Http\Response())
+              ->withHeader("Content-Type", "text/plain")
+              ->withStatus(200)
+              ->withBody(
+                  new \LORIS\Http\StringStream(
+                      json_encode($this->getUploadDocFields($record))
+                  )
+              );
+            break;
+        default:
+            return (new \LORIS\Http\Response())
+                ->withHeader("Content-Type", "text/plain")
+                ->withStatus(405)
+                ->withHeader("Allow", "GET,POST,PUT,DELETE");
+        }
+    }
+    /**
+     * Handles the document editing process
+     *
+     * @throws \DatabaseException
+     *
+     * @return void
+     */
+    function editDocFile(): void
+    {
+        $db   = \Database::singleton();
+        $user = \User::singleton();
+        $req  = array();
+        $this->parseRawHttpRequest($req);
+        $updateValues = [
+                         'instrument' => $req['instrument'] ?? null,
+                         'comments'   => $req['comments'] ?? null,
+                         'version'    => $req['version'] ?? null,
+                         'visitLabel' => $req['visitLabel'] ?? null,
+                         'pscid'      => $req['pscid'] ?? null,
+                        ];
+        try {
+            $db->update(
+                'document_repository',
+                $updateValues,
+                ['record_id' => $req['id']]
+            );
+        } catch (\DatabaseException $e) {
+            showError("Could not update the file. Please try again!");
+        }
+    }
+    /**
+     * Handles the raw form-data from PUT method to an array.
+     *
+     * @param array $a_data an init array
+     *
+     * @throws \DatabaseException
+     * @return void
+     */
+    function parseRawHttpRequest(array &$a_data): void
+    {
+        // read incoming data
+        $input = file_get_contents('php://input');
+        // grab multipart boundary from content type header
+        preg_match('/boundary=(.*)$/', $_SERVER['CONTENT_TYPE'], $matches);
```

Comment:
This is getting really close to mergeable and this is the only problem I have with it now. I'm not sure what you're trying to do here, but any PHP superglobals (ie. `$_SERVER`) should be better accessed indirectly via the PSR7 request objects.